### PR TITLE
Use peer dependencies

### PR DESCRIPTION
**Pull request by mojito317**
## Checklist

- [x] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](../blob/master/DCO1.1.txt)
- [x] Added tests for code changes _or_ test/build only changes
- [x] Updated the change log file (`CHANGES.md`|`CHANGELOG.md`) _or_ test/build only changes
- [x] Completed the PR template below:

## Description

I moved `ibm-cloud-sdk-core` and `retry-axios` to the `peerDependencies` section of the `package.json` file, because those are installed from core or cloudant-node-sdk and no need for duplicates.

I used v12.22.10 of Node to regenerate the `package-lock.json` file, because with `16` the [builds failed](https://cloudant-sdks-jenkins.swg-devops.com/job/Main/job/couchbackup/job/use-peerDependencies/1/display/redirect).

I picked the minimum version based on these:

- The version for `ibm-cloud-sdk-core` in `@ibm-cloud/cloudant@0.0.25` is [2.17.10](https://github.com/IBM/cloudant-node-sdk/blob/553a6d1eb50e989aa84186ab8b3c6a2bea12e2ff/package.json#L40). 
- The version for `retry-axios` in  `ibm-cloud-sdk-core@2.17.10` is [^2.6.0](https://github.com/IBM/node-sdk-core/blob/3f19fce13f2b8a3d8c999b441b07002e16fc8a1d/package.json#L96). `2.6.0` is the latest version as well, so I used that one.

## Approach

I moved `ibm-cloud-sdk-core` and `retry-axios` to the `peerDependencies` section of the `package.json` file and updated `package-lock.json` from a v12.22.10 of Node environment.

## Schema & API Changes

- "No change"

## Security and Privacy

- N/A build or packaging only changes

## Monitoring and Logging

- "No change"

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # Unreleased
+- [NOTE] Put `retrty-axios` and `ibm-cloud-sdk-core` as peer dependencies.
```
Comment (ricellis) [inline CHANGES.md:2]:
```suggestion
- [NOTE] Move `retry-axios` and `ibm-cloud-sdk-core` to peerDependencies.
```

File: package.json [hunk 2]
```diff
@@ -20,14 +20,16 @@
...
     "@ibm-cloud/cloudant": "0.0.25",
-    "retry-axios": "2.x",
     "async": "^3.1.0",
     "commander": "^9.0.0",
     "debug": "~4.3.2",
     "tmp": "0.2.1"
   },
+  "peerDependencies": {
+    "ibm-cloud-sdk-core": ">=2.17.10",
+    "retry-axios": ">=2.6.0"
```
Comment (ricellis) [inline package.json:31]: To match the exact install range used by the core itself:
```suggestion
    "retry-axios": "^2.6.0"
```

**Review by ricellis — APPROVED** [review]
+1 just a couple nits

**Review by tomblench — APPROVED** [review]

**Review by eiri — APPROVED** [review]

**Comment by mojito317** [comment]
Thanks for the reviews! I am about to squash and merge this!

**Pull request merged**